### PR TITLE
Search question

### LINKED ISSUE #11: As a user, I want to search a question, so that I can find already answered questions.

### PR DESCRIPTION
**Pull request by LarsvdBrandt**

### CODE REVIEW

File: Abracadabra/abracadabra_web_app/src/components/QuestionBody.js [hunk 1]
```diff
@@ -1,55 +1,64 @@
...
+  if (question.length > 0) {
+    return (
+      <div className="container mt-5">
+        <h1>{message}</h1>
+        {question.map((question) => (
+          <div className="card mt-1" key={question.id}>
+            <div className="BodyQuestion-CardBody">
+              <a
+                href={
+                  "http://localhost:3000/subject/" +
```
Comment (RealSnowKid) [inline Abracadabra/abracadabra_web_app/src/components/QuestionBody.js]: Should be just `/subject/` without the localhost.

```suggestion
                  "/subject/" +
```

File: Abracadabra/abracadabra_web_app/pages/subject/[subject]/question/[id].js [hunk 2]
```diff
@@ -1,106 +1,102 @@
...
+  // const { subject, id } = router.query
+  const [question, setQuestion] = useState({});
+  const [answers, setAnswers] = useState([]);
 
-        // get the subject from router, to pass to the navbar for navigation button and title
-        // const router = useRouter()
-        // const { subject, id } = router.query
-        const [question, setQuestion] = useState({});
-        const [answers, setAnswers] = useState([]);
+  console.log(data);
```
Comment (RealSnowKid) [inline Abracadabra/abracadabra_web_app/pages/subject/[subject]/question/[id].js]: console.log

```suggestion

```

File: Abracadabra/abracadabra_web_app/pages/subject/[subject]/search/[search].js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+import QuestionBody from "src/components/QuestionBody.js";
+import DefaultErrorPage from "next/error";
+import SubjectService from "src/services/SubjectService";
+import { useRouter } from "next/router";
+
+function Subject({ subjectName, response }) {
+  const router = useRouter();
+  const { subject, search } = router.query;
+  console.log(search);
+  console.log(response);
```
Comment (RealSnowKid) [inline Abracadabra/abracadabra_web_app/pages/subject/[subject]/search/[search].js]: `console.log()`s

```suggestion

```

**Review by RealSnowKid — CHANGES_REQUESTED** [review]
Some little changes with `console.log()` still left in code and a `href` typo.

**Review by RealSnowKid — APPROVED** [review]

File: Abracadabra/abracadabra_web_app/pages/subject/[subject]/search/[search].js [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  const router = useRouter();
+  const { subject, search } = router.query;
+
+  if (response === 404 || response == "failure" || response === 400) {
+    return <DefaultErrorPage statusCode={404} />;
+  }
+
+  return (
+    <>
+      <Navbar subjectTitle={subjectName} />
```
Comment (TijmenC) [inline Abracadabra/abracadabra_web_app/pages/subject/[subject]/search/[search].js:18]: Possible to make it dynamic in the future? This basically is using a seperate page for showing the response

File: Abracadabra/abracadabra_web_app/src/components/Navbar.js [hunk 5]
```diff
@@ -32,29 +46,66 @@ function Navbar({subjectTitle}) {
...
+        </a>
+        <button
+          className="searchMobile mx-auto"
+          onClick={handleShow}
+          id="mobile_search"
+        >
+          <FontAwesomeIcon icon={faSearch} />
+        </button>
 
         {/* <a className="navbarHubIcon" href="/loginpage"><FontAwesomeIcon icon={faUserCircle} /></a> */}
```
Comment (TijmenC) [inline Abracadabra/abracadabra_web_app/src/components/Navbar.js:105]: Outcommented code not deleted

File: Abracadabra/abracadabra_web_app/src/components/QuestionBody.js [hunk 6]
```diff
@@ -1,55 +1,64 @@
...
+          <div className="card mt-1" key={question.id}>
+            <div className="BodyQuestion-CardBody">
+              <a
+                href={
+                  "/subject/" +
+                  subject +
+                  "/question/" +
+                  question.id
+                }
+                style={{
```
Comment (TijmenC) [inline Abracadabra/abracadabra_web_app/src/components/QuestionBody.js:26]: Don't put styling in the HTML code make a seperate .css class

File: Abracadabra/abracadabra_web_app/src/components/Navbar.js [hunk 7]
```diff
@@ -16,11 +23,18 @@ function Navbar({subjectTitle}) {
...
       handleOpen();
     }
   }
+  const handleChange = (event) => {
+    setSearchString({
+      ...searchString,
+      [event.target.name]: event.target.value,
+    });
+
+    console.log(searchString);
```
Comment (TijmenC) [inline Abracadabra/abracadabra_web_app/src/components/Navbar.js:36]: Remove console.log

File: Abracadabra/abracadabra_web_app/src/components/Navbar.js [hunk 8]
```diff
@@ -32,29 +46,66 @@ function Navbar({subjectTitle}) {
...
+        <a className="navbarHubIcon" href="/">
+          <FontAwesomeIcon icon={faHome} />
+        </a>
         <div className="navbarCollapse d-flex mx-auto">
-          <a className="navbar-brand d-flex" href={'/subject/' + subjectTitle}>
+          <a className="navbar-brand d-flex" href={"/subject/" + subjectTitle}>
             <img src={logo} className="navbarLogo" />
             <h1 className="navbarText mt-2">{subjectTitle}</h1>
           </a>
           {/* <div className="search-container align-self-center"> */}
```
Comment (TijmenC) [inline Abracadabra/abracadabra_web_app/src/components/Navbar.js:57]: Outcommented code

File: Abracadabra/abracadabra_web_app/src/components/Navbar.js [hunk 9]
```diff
@@ -16,11 +23,18 @@ function Navbar({subjectTitle}) {
   function handleClick(e) {
     if (!e.target.closest("#dropdown") && open) {
       handleOpen();
-    }
-    else if (!e.target.closest("#dropdown_item") && open) {
+    } else if (!e.target.closest("#dropdown_item") && open) {
```
Comment (TijmenC) [inline Abracadabra/abracadabra_web_app/src/components/Navbar.js:26]: Remove old code for dropdown

File: Abracadabra/abracadabra_web_app/src/components/Navbar.js [hunk 10]
```diff
@@ -32,29 +46,66 @@ function Navbar({subjectTitle}) {
...
-          <a href={'/subject/' + subjectTitle + '/createquestion'} className="btn btn-outline-secondary navbarButton align-self-center ml-2">Ask a Question</a>
-          </div>
+          <a
+            href={"/subject/" + subjectTitle + "/createquestion"}
+            className="btn btn-outline-secondary navbarButton align-self-center ml-2"
+          >
+            Ask a Question
+          </a>
+        </div>
         {/* </div> */}
```
Comment (TijmenC) [inline Abracadabra/abracadabra_web_app/src/components/Navbar.js:93]: Outcommented code

**Pull request merged**